rtbkit/core/router/filter_pool.cc

```diff
@@ -116,9 +132,14 @@ filter(const BidRequest& br, const ExchangeConnector* conn, const ConfigSet& mas
 
         if (sampleStats) {
             ticksStart = recordTime(ticksStart, filter);
-            recordDiff(current, filter, configs ^ filtered);
+            if ( filter->name() != SegmentsFilter::name ){
```

Comment:
With the above suggestions, this if is no longer necessary. Instead, I would always call `recordDiff` to avoid breaking backwards compatibility and call `recordReason` whenever the reason map is not empty in FilterState.